### PR TITLE
Draft: Release blog post

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #824

Filing here to facilitate collaboration -- I've only got the first paragraph or so going.

### CODE REVIEW

File: linter-300.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+style, readability, default, and configurable. Each tag gets its own documentation
+page (e.g. `?readability_linters`) which describes the tag and lists all of the
+associated linters exported by lintr.
+
+To further facilitate discoverability, we've added `available_linters()` and
+`available_tags()` to list registered linters (including custom linters not
+exported by lintr itself; see `?available_linters`).
+
+Moreover, lintr's documentation is now available as a website thanks to
+Hadley Wickham's contribution to create a pkgdown website for the package: lintr.r-lib.org.
```
Comment (AshesITR) [inline linter-300.md:75]: Make it a link?

Comment (MichaelChirico) [reply linter-300.md:75]: Oh, I assumed it would auto-link...

**Comment by MichaelChirico** [comment]
BTW, we won't merge this I guess. Just keep it in this branch for review and then pass the source along to Hadley to host it on the tidyverse blog. IIUC that should come after releasing to CRAN so that readers can install right away.

**Comment by MichaelChirico** [comment]
I was thinking about a section on the conversion to XPath logic / export of `xml_nodes_to_lints()`, WDYT @AshesITR, worthy of inclusion? Want to take a stab at it?

File: linter-300.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+See the NEWS for the complete list.
+
+## Other improvements
+
+This is a big release -- almost 2 years in the making -- and there has been a plethora of smaller
+but nonetheless important changes to lintr. Please check the NEWS for a complete enumeration of these;
+here are a few more highlights:
+
+# What's next in lintr
```
Comment (AshesITR) [inline linter-300.md:107]: heading structure seems less than ideal.
Add a level 1 heading at the very top, or increase all levels down from here by one?

Comment (MichaelChirico) [reply linter-300.md:113]: maybe it was just missing an L1 header for "Release highlights"? added now

**Comment by AshesITR** [comment]
I'd leave out the linter development stuff (including `is_lint_level()`) and move `use_lintr()` further up so new users can easily see how to try out the new package.
I think the audience will contain far more end users than linter developers.

Maybe include the snippets from `README.md` regarding using lintr?

**Comment by MichaelChirico** [comment]
Highlighted all the end-user-oriented linters in the beginning, WDYT?

**Review by AshesITR — DISMISSED** [review]
Symbolic approval - the Blog post looks excellent now.
Thanks for the hard work!

**Comment by MichaelChirico** [comment]
great! @jimhester @russHyde @renkun-ken do you want to have a read through?

File: linter-300.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+Consider the following example:
+
+```r
+T_and_F_symbol_linter=function(){
+  list()
+}
+```
+
+This snippet generates 5 lints: `object_name_linter()` because of the uppercase `T` and `F`;
```
[the remark this answers is not in the document]
Comment (MichaelChirico) [reply linter-300.md]: great idea! done

File: linter-300.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+   by identifying usages of `data.frame()` that (1) have known string columns and (2) don't declare
+   a value for `stringsAsFactors`, and thus rely on the R version-dependent default.
+
+See the [NEWS](https://lintr.r-lib.org/news/index.html#google-linters-development-version) for the complete list.
+
+## Other improvements
+
+This is a big release -- almost 2 years in the making -- and there has been a plethora of smaller
+but nonetheless important changes to lintr. Please check the
+[NEWS](https://lintr.r-lib.org/news/index.html#lintr-development-version)
```
Comment (MichaelChirico) [inline linter-300.md:162]: NB: we need to change this anchor once the NEWS updates to have a 3.0.0 anchor

**Comment by MichaelChirico** [comment]
@hadley let us know anything else we should do / how we should proceed from here after 3.0.0 hits CRAN

**Comment by jimhester** [comment]
This LGTM, thank you for writing it. Where were you planning on posting this, on tidyverse.org or elsewhere? Or just on the pkgdown site directly?

**Comment by russHyde** [comment]
LGTM. Thanks so much for your hard work on lintr

**Comment by MichaelChirico** [comment]
> This LGTM, thank you for writing it. Where were you planning on posting this, on tidyverse.org or elsewhere? Or just on the pkgdown site directly? For lintr 2.0.0 I had a post on my personal blog (https://www.jimhester.com/post/lintr-2-0-0/), but that doesn't seem the best place for this

Hadley had offered to host it on the tidyverse blog: https://twitter.com/hadleywickham/status/1528081727259824128

**Comment by MichaelChirico** [comment]
@jimhester I like the idea of including people who opened issues, not just those who changed code -- do you have a snippet handy for calculating that? Otherwise I guess it's just `issues |> filter(dates_since(prev_release)) |> unique(OP)` aided by {gh}?

**Comment by hadley** [comment]
@MichaelChirico `usethis::use_tidy_thanks()` might do what you need

**Comment by jimhester** [comment]
@MichaelChirico thanks again for writing this up, I think it sounds great!

The next step would be to open this as a pull request on the tidyverse blog, at https://github.com/tidyverse/tidyverse.org.

I used https://unsplash.com/photos/NPPNHZK1U0s for the picture in the 2.0.0 release, you are welcome to reuse that, or pick another one as you prefer.

**Comment by AshesITR** [comment]
Dropping this here. Can we close this PR, @MichaelChirico?
https://github.com/tidyverse/tidyverse.org/pull/584

**Comment by MichaelChirico** [comment]
I would leave it open as a TODO reminder for now

**Comment by MichaelChirico** [comment]
Closing -- tidyverse blog link for reference: https://www.tidyverse.org/blog/2022/07/lintr-3-0-0/

**Pull request closed without merging**